src/main/kotlin/com/terraformation/backend/search/table/DocumentsTable.kt

```diff
@@ -0,0 +1,56 @@
+package com.terraformation.backend.search.table
+
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.tables.references.PROJECTS
+import com.terraformation.backend.db.docprod.DocumentId
+import com.terraformation.backend.db.docprod.DocumentTemplateId
+import com.terraformation.backend.db.docprod.tables.references.DOCUMENTS
+import com.terraformation.backend.db.docprod.tables.references.DOCUMENT_SAVED_VERSIONS
+import com.terraformation.backend.search.SearchTable
+import com.terraformation.backend.search.SublistField
+import com.terraformation.backend.search.field.SearchField
+import org.jooq.Condition
+import org.jooq.OrderField
+import org.jooq.Record
+import org.jooq.TableField
+import org.jooq.impl.DSL
+import org.jooq.impl.SQLDataType
+
+class DocumentsTable(tables: SearchTables) : SearchTable() {
+  override val primaryKey: TableField<out Record, out Any?>
+    get() = DOCUMENTS.ID
+
+  override val sublists: List<SublistField> by lazy {
+    with(tables) {
+      listOf(projects.asMultiValueSublist("projects", PROJECTS.ID.eq(DOCUMENTS.PROJECT_ID)))
+    }
+  }
+
+  override val fields: List<SearchField> =
+      listOf(
+          timestampField("createdTime", DOCUMENTS.CREATED_TIME, nullable = false),
+          idWrapperField("documentTemplateId", DOCUMENTS.DOCUMENT_TEMPLATE_ID) {
+            DocumentTemplateId(it)
+          },
+          idWrapperField("id", DOCUMENTS.ID) { DocumentId(it) },
+          integerField(
```

Comment:
Should this be an `idWrapperField` instead? Then you don't need to cast to an integer.

Reply:
I did try that but ended up with type errors.

![SCR-20240807-knyr.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/wJnPX9wLZAkiGXEBFJxL/540c4a87-b474-406f-adaf-04307902b1c7.png)

Reply:
Ah, okay, let me look at that and see if I can make it not require a `TableField`.

Reply:
Looks like there's no reason to require a `TableField`. To make this work you can edit `IdWrapper.kt` to change `TableField<*, T?>` to `Field<T?>` (make sure you import the `Field` class from the jOOQ library, not the one from the Java standard library).

Reply:
I was able to make it work in my latest commit

Reply:
Sweet, that's what I went for!